BlockchainSdk/Blockchains/Polkadot/AccountHealthCheck/Network/SubscanPolkadotAccountHealthNetworkService.swift

```diff
@@ -0,0 +1,227 @@
+//
+//  SubscanPolkadotAccountHealthNetworkService.swift
+//  BlockchainSdk
+//
+//  Created by Andrey Fedorov on 25.03.2024.
+//  Copyright © 2024 Tangem AG. All rights reserved.
+//
+
+import Foundation
+import enum Moya.MoyaError
+
+public final class SubscanPolkadotAccountHealthNetworkService {
+    private typealias SkipRetryIf = (_ error: Error) -> Bool
+
+    private let provider = NetworkProvider<SubscanAPITarget>()
+    private let encoder: JSONEncoder
+    private let decoder: JSONDecoder
+    private let isTestnet: Bool
+    private let pageSize: Int
+
+    public init(
+        isTestnet: Bool,
+        pageSize: Int
+    ) {
+        self.isTestnet = isTestnet
+        self.pageSize = pageSize
+
+        encoder = JSONEncoder()
+        encoder.keyEncodingStrategy = .convertToSnakeCase
+
+        decoder = JSONDecoder()
+        decoder.keyDecodingStrategy = .convertFromSnakeCase
+    }
+
```

Comment:
Тоже больше предложение, чем крит
Так кажется наглядней просто все опции энкодера/декодера
а то мейби в`init` потеряются

```suggestion
    private let encoder: JSONEncoder = {
            let encoder = JSONEncoder()
            encoder.keyEncodingStrategy = .convertToSnakeCase
            return encoder
    }()
    private let decoder: JSONDecoder = {
            let decoder = JSONDecoder()
            decoder.keyEncodingStrategy = .convertToSnakeCase
            return decoder
    }()
    private let isTestnet: Bool
    private let pageSize: Int

    public init(
        isTestnet: Bool,
        pageSize: Int
    ) {
        self.isTestnet = isTestnet
        self.pageSize = pageSize
    }
```

Reply:
Логично, сделал так